front/src/modules/timesStops/TimesStops.tsx

```diff
@@ -1,72 +1,33 @@
-import { useState, useEffect } from 'react';
-
 import cx from 'classnames';
-import { isEqual } from 'lodash';
 import { DynamicDataSheetGrid, type DataSheetGridProps } from 'react-datasheet-grid';
+import type { Operation } from 'react-datasheet-grid/dist/types';
 import { useTranslation } from 'react-i18next';
 
-import { useOsrdConfActions } from 'common/osrdContext';
-import type { IsoDateTimeString } from 'common/types';
-import { isVia } from 'modules/pathfinding/utils';
-import type { SuggestedOP } from 'modules/trainschedule/components/ManageTrainSchedule/types';
-import type { PathStep } from 'reducers/osrdconf/types';
-import { useAppDispatch } from 'store';
-
-import {
-  formatSuggestedViasToRowVias,
-  updateRowTimesAndMargin,
-  updateDaySinceDeparture,
-  durationSinceStartTime,
-} from './helpers/utils';
 import { useTimeStopsColumns } from './hooks/useTimeStopsColumns';
-import { TableType } from './types';
-import type { PathWaypointRow } from './types';
+import { type TableType, type TimeStopsRow } from './types';
 
-export const WITH_KP = true;
-
-type TimesStopsProps = {
-  allWaypoints?: SuggestedOP[];
-  pathSteps?: PathStep[];
-  startTime?: IsoDateTimeString;
+type TimesStopsProps<T extends TimeStopsRow> = {
+  rows?: T[];
   tableType: TableType;
   cellClassName?: DataSheetGridProps['cellClassName'];
   stickyRightColumn?: DataSheetGridProps['stickyRightColumn'];
   headerRowHeight?: number;
+  onChange?: (newRows: T[], op: Operation) => void;
 };
 
-const TimesStops = ({
-  allWaypoints,
-  pathSteps,
-  startTime,
+const TimesStops = <T extends TimeStopsRow>({
+  rows,
   tableType,
   cellClassName,
   stickyRightColumn,
   headerRowHeight,
-}: TimesStopsProps) => {
-  const isInputTable = tableType === TableType.Input;
+  onChange,
+}: TimesStopsProps<T>) => {
   const { t } = useTranslation('timesStops');
 
-  const dispatch = useAppDispatch();
-  const { upsertSeveralViasFromSuggestedOP } = useOsrdConfActions();
-
-  const [rows, setRows] = useState<PathWaypointRow[]>([]);
+  const columns = useTimeStopsColumns(tableType, rows);
```

Comment:
you have a 
`Argument of type 'T[] | undefined' is not assignable to parameter of type 'SuggestedOP[] | undefined'.
`
perhaps make useTimeStopColumns a generic function too

Reply:
Done :)